### PR TITLE
Feature/collaboration scope

### LINKED ISSUE #245: [Feature Request] Extend API security layers with Collaboration scope
**Problem description**
Currently it is not possible to edit settings on a Collaboration level. In practice this is usually needed and currently these kind of collaboration admins receive too much permission on the server.

**Desired solution**
Currently the main scopes we are using are `organization` and `global`. The idea is to add a scope `collaboration` for all endpoints. The main advantage is that non-superadmin users can manage collaborations without seeing everything on the server

### PR DESCRIPTION
**Pull request by bartvanb**
Fix #245 and #635

### CODE REVIEW

**Comment by bartvanb** [comment]
Ignore codacy

**Review by frankcorneliusmartin — CHANGES_REQUESTED** [review]

File: vantage6-server/vantage6/server/permission.py [hunk 1]
```diff
@@ -44,6 +47,182 @@ def add(self, scope: Scope, operation: Operation) -> None:
...
+            operation
+
+        Returns
+        -------
+        bool
+            True if the operation is allowed on the organization, False
+            otherwise
+        """
+        # check if the entity has global permission
+        global_perm = getattr(self, f'{operation.value}_{Scope.GLOBAL.value}')
```
[the remark this answers is not in the document]
Comment (bartvanb) [reply vantage6-server/vantage6/server/permission.py:72]: Yes I added `str` to the Enum (both Operation and Scope) and checked that it works

File: vantage6-server/vantage6/server/resource/__init__.py [hunk 2]
```diff
@@ -267,3 +285,37 @@ def parse_datetime(dt: str = None, default: datetime = None) -> datetime:
...
+        collaboration. If not given, return all organization ids of all
+        collaborations the user or node is in.
+
+    Returns
+    -------
+    list[int]
+        List of organization ids
+    """
+    if collab_id:
+        return [
```
[the remark this answers is not in the document]
Comment (bartvanb) [reply vantage6-server/vantage6/server/resource/__init__.py:310]: This function is not really used for permission checks. An example where it is used: when someone requests roles with `collaboration_id=x`, this function is used to determine which organization ids have to be put in the query filter.

File: vantage6-server/vantage6/server/permission.py [hunk 3]
```diff
@@ -44,6 +47,182 @@ def add(self, scope: Scope, operation: Operation) -> None:
...
+        Check if the user or node can perform the operation on a certain
+        collaboration
+
+        Parameters
+        ----------
+        operation: Operation
+            Operation to check if allowed
+        collaboration_id: int
+            Collaboration id on which the operation should be allowed
+        auth: Authenticatable
```
[the remark this answers is not in the document]
Comment (bartvanb) [reply vantage6-server/vantage6/server/permission.py:108]: True, I made a utils function to get auth's collaborations and also did the same for organizations
Indeed this makes endpoints much cleaner again :-)

**Review by frankcorneliusmartin — APPROVED** [review]

**Pull request merged**